lib/modules/manager/poetry/extract.ts

```diff
@@ -12,38 +12,36 @@ import * as pep440Versioning from '../../versioning/pep440';
 import * as poetryVersioning from '../../versioning/poetry';
 import type { PackageDependency, PackageFile } from '../types';
 import { extractLockFileEntries } from './locked-version';
-import type { PoetryFile, PoetrySection } from './types';
+import type { PoetryDependency, PoetryFile, PoetrySection } from './types';
 
 function extractFromSection(
   parsedFile: PoetryFile,
   section: keyof PoetrySection,
   poetryLockfile: Record<string, string>
 ): PackageDependency[] {
-  const deps = [];
-  const sectionContent = parsedFile.tool.poetry[section];
+  const deps: PackageDependency[] = [];
+  const sectionContent = parsedFile.tool?.poetry[section] ?? null;
```

Comment:
```suggestion
  const sectionContent = parsedFile.tool?.poetry[section];
```
i don't think we need the fallback, the next if will catch undefined too